glusterd2/commands/devices/adddevice.go

```diff
@@ -0,0 +1,71 @@
+package devicecommands
+
+import (
+	"net/http"
+
+	device "github.com/gluster/glusterd2/glusterd2/device"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/pkg/api"
+
+	"github.com/pborman/uuid"
+)
+
+func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	req := new(api.AddDeviceReq)
+	if err := restutils.UnmarshalRequest(r, req); err != nil {
+		logger.WithError(err).WithField("devices", "devices").Error("Failed to marshal request")
```

Comment:
The `WithField` doesn't make any sense here as both are same string (key and the value). The error message is also incorrect. The request is bring unmarshalled.